lib/utils.js

```diff
@@ -187,6 +203,38 @@ var normalizeHintField = function normalizeHintField(hint) {
   return finalHint;
 };
 
+const commandOptions = [
```

Comment:
`commandOptions` for what? Now that this is in the scope of the `utils.js`, this name is no longer sufficiently specific IMO.

Reply:
I refactored this so it is now explicitly using a [validIndexOptions](https://github.com/mongodb/node-mongodb-native/pull/2326/files#diff-aa40aa8bcfc7826a48153f27cc6a7c7cR10) `Set` to merge the options into the index spec.